content/en/blog/2022-03-21-tools-to-manage-helm-declaratively.md

```diff
@@ -0,0 +1,75 @@
+
+There are some differences between the tools we've looked at so far. The following table provides some insight into their differences. This is not exhaustive and you should evaluate any tools you use yourself.
+
+| | Retains Helm release info | Supports Helm hooks | OCI support |
+| -- | -- | -- | -- |
+| Helmfile | ✅ | ❓[^1] | 🚫 |
+| Terraform Helm provider | ✅ | ✅[^2] | 🚫[^3] |
+| Captain Helm controller | ✅ | ❓[^4] | 🚫[^5] |
+| Flux Helm controller | ✅ | ✅  | 🚫[^6] |
+| ArgoCD | 🚫 | 🚫[^7] | ✅[^8] |
+| Orkestra | ❓ | ❓ | ❓ |
+
+[^1]: Has a custom concept of hooks, not necessarily mapped to Helm hooks. See readme [hooks section](https://github.com/roboll/helmfile#hooks) and [this issue](https://github.com/roboll/helmfile/issues/1291) for clarification and work in progress.
+[^2]: Note there are [some issues](https://github.com/hashicorp/terraform-provider-helm/issues/683) with Helm hooks and wait configurations
+[^3]: See hashicorp/terraform-provider-helm issues [#633](https://github.com/hashicorp/terraform-provider-helm/issues/633), [#666](https://github.com/hashicorp/terraform-provider-helm/issues/666), [#827](https://github.com/hashicorp/terraform-provider-helm/issues/827), [#765](https://github.com/hashicorp/terraform-provider-helm/issues/765), and [#655](https://github.com/hashicorp/terraform-provider-helm/issues/655).
+[^4]: This is not a very widely used project, so unclear which or how well Helm hooks are supported. Scott may want to have another chat with the maintainer (TO-DO: fold in this [reminder code](https://github.com/alauda/captain/blob/master/pkg/helm/printer.go#L23) in a simple way)
+[^5]: Relies on a related project [alauda/oci-chartrepo](https://github.com/alauda/oci-chartrepo) to mix concepts of using oci registry as helm chart repo
+[^6]: Because Flux makes full use of the Helm SDK, as of Helm v3.8.0 Flux is now unblocked to add OCI artifact integration (Flux team members helped finish bringing OCI support out of experimental into a full feature in Helm). Work on this is in progress for Flux. You can follow this [canonical issue](https://github.com/fluxcd/source-controller/issues/124) for progress. See the current [design doc](https://hackmd.io/HS9ZqSCWQNqSefdaWFylJQ?view) for details.
+[^7]: Because Argo does not retain Helm release information, there is an [attempt to map](https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks) Helm hooks to ArgoCD hooks, however, there are far fewer Argo hooks and unmappable concepts such as no differentiation between install and upgrade. You can work around this by writing your charts specifically for ArgoCD, however hooks in commonly used community charts will not work.
+[^8]: ArgoCD shells out to the Helm CLI, only to render templates. This has allowed Argo to turn on Helm CLI's OCI feature before it was finished, for the same reason that it can not support Helm features beyond templating. Because of this, OCI is not part of the ArgoCD source architecture.
+
```

Comment:
```suggestion
[^9] See readme [OCI registries section](https://github.com/roboll/helmfile#oci-registries) for details.

```

Reply:
See [comment](https://github.com/helm/helm-www/pull/1293#discussion_r831602302) above 🙂

Reply:
Resolving in  https://github.com/scottrigby/helm-www/pull/1